src/renderer_vk.cpp

```diff
@@ -9479,11 +9475,13 @@ VK_DESTROY
 		{
 			BGFX_PROFILER_SCOPE("scratchBuffer::flush", kColorResource);
 			scratchBuffer.flush();
+			scratchBuffer.reset();
```

Comment:
Is there reason to not have `flush` inside `reset`?

Reply:
Note that in the master branch, flush() was being called at the right time, but after the reset(). This effectively made flush() a no-op, because we only try to flush the memory region that we touched. However, I guess we were lucky, and it did always update?

Reply:
> They are two separate things IMO. Currently they are used one after the other, indeed.

I understand this, but I see case where `flush` is called independently from `reset` but it looks like calling `reset` without `flush` first would be a bug always? Is there case where calling `reset` without `flush` would be desirable?

Reply:
Merged them into one function.